aten/src/ATen/native/ts_native_functions.yaml

```diff
@@ -12,14 +12,14 @@ full_codegen:
   - addcdiv
```

Comment:
cc @wconstab @alanwaketan @JackCaoG @antoniojkim

This PR will allow LTC to not have to worry about directly lowering inplace/out= operators. The high level idea is that:
- every inplace/out= operator has a functional variant now (e.g. `random_` has `random.functional`)
- LTC can lower the functional variant of every op, and rely on existing codegen to generate kernels for `random_` and `random.out`

One consequence though is that if you have some existing inplace/out= ops that you are directly lowering, you need to either:
(1) put it in the `supported` list instead of the `codegen` list in your yaml file
(2) put it in the `codegen` list to get full codegen support, but update your lowering logic to handle the functional variant of the operator instead

Reply:
does this effectively make `lazy_view.cpp` not do anything? As in its there but no longer used?

Reply:
oh nope, this PR doesn't change the behavior of view ops for LTC at all - it will still do all of the `lazy_view.cpp` stuff that it does today to keep track of aliasing.

The plan is to eventually kill that code though - I'm working on integrating LTC with the version of functionalization that we have in core. It will basically mean that:
- All of the aliasing machinery in LTC can be removed
- Backends to LTC will need to write lowering for new `{view}_copy` operators. These are basically functional equivalents of the existing view operators that don't perform any aliasing. E.g. `view_copy(Tensor, int[] sizes, int[] strides) -> Tensor`

Reply:
I'm not sure why any lowerings would be impacted; the IR itself is always 'functional' and the lowerings act on the IR.  Right, or am I missing something?  I think only the bindings (aten nativefuncs that build IRs) would be impacted.

Reply:
Yeah that's right - the only integration churn I was thinking of was that if XLA/Cerebras are already relying on the existing LTC codegen for inplace ops, this PR would require them to change them to be functional ops (which should be a simple change like you pointed out).

Reply:
edit: actually it turns out that lowering the functional `arange` variant through codegen is non-trivial and breaks (from the failing CI), so I'm just going to update it to let the codegen keep working with `arange.start_out`. I can keep all of the other changes where I remove inplace variants though.

Reply:
@bdhirsh does that means pytorch/xla should remove all of the manually lowering of `in_place` and `out` ops?

Reply:
yep - e.g. instead of lowering both `addcdiv_` and `addcdiv`, XLA can just lower `addcdiv` (so you can delete the redundant inplace lowering).

(Although I updated the PR to make that not required - you should be able to still lower out= and inplace ops through the LTC codegen if you want to)

Reply:
@bdhirsh Given that 1.12 branch cut is very close, do you plan to merge this pr to the pytoch 1.12?

Reply:
I think this PR is actually pretty low risk - the bulk of the PR is just adding functionality for functionalization, which isn't heavily depended on as of the previous release. I think the actual changes to LTC codegen are pretty small and non-disruptive.

The changes to the set of aten ops lowered by the TS plugin also seems low risk (lowering a few functional variants instead of their inplace variants, and relying on codegen support). But let me know if there's anything you're worried about.

Reply:
If I understand correctly I need to have a follow up pr on pt/xla end. Trying to understand the plan of merging this pr so I can plan accordingly on the pt/xla side.

Reply:
I don't think any changes are needed to the tip of pt/xla master.

If you're actively trying to move more lowerings to use the LTC codegen (by moving it under the "codegen" section of the yaml, then I think basically the only change that you should need is if you have redundant lowerings both a functional + inplace operator:

If before you had:
```
codegen:
- celu
- celu_
- bernoulli
- bernoulli_.Tensor
```

Then update to:
```
codegen:
- celu
- bernoulli
```

Reply:
sounds good. FYI @miladm and @wonjoolee95